kafka-rest/src/main/java/io/confluent/kafkarest/KafkaRestConfig.java

```diff
@@ -327,309 +327,279 @@
-            ConfigDef.Importance.HIGH,
-            KAFKACLIENT_SSL_KEY_PASSWORD_DOC
-        )
-        .define(
-            KAFKACLIENT_SSL_ENABLED_PROTOCOLS_CONFIG,
-            ConfigDef.Type.STRING,
-            "TLSv1.2,TLSv1.1,TLSv1",
-            ConfigDef.Importance.MEDIUM,
-            KAFAKSTORE_SSL_ENABLED_PROTOCOLS_DOC
-        )
-        .define(
-            KAFKACLIENT_SSL_PROTOCOL_CONFIG,
-            ConfigDef.Type.STRING,
-            "TLS",
-            ConfigDef.Importance.MEDIUM,
-            KAFAKSTORE_SSL_PROTOCOL_DOC
-        )
-        .define(
-            KAFKACLIENT_SSL_PROVIDER_CONFIG,
-            ConfigDef.Type.STRING,
-            "",
-            ConfigDef.Importance.MEDIUM,
-            KAFAKSTORE_SSL_PROVIDER_DOC
-        )
-        .define(
-            KAFKACLIENT_SSL_CIPHER_SUITES_CONFIG,
-            ConfigDef.Type.STRING,
-            "",
-            ConfigDef.Importance.LOW,
-            KAFKACLIENT_SSL_CIPHER_SUITES_DOC
-        )
-        .define(
-            KAFKACLIENT_SSL_ENDPOINT_IDENTIFICATION_ALGORITHM_CONFIG,
-            ConfigDef.Type.STRING,
-            "",
-            ConfigDef.Importance.LOW,
-            KAFKACLIENT_SSL_ENDPOINT_IDENTIFICATION_ALGORITHM_DOC
-        )
-        .define(
-            KAFKACLIENT_SASL_KERBEROS_SERVICE_NAME_CONFIG,
-            ConfigDef.Type.STRING,
-            "",
-            ConfigDef.Importance.MEDIUM,
-            KAFKACLIENT_SASL_KERBEROS_SERVICE_NAME_DOC
-        )
-        .define(
-            KAFKACLIENT_SASL_MECHANISM_CONFIG,
-            ConfigDef.Type.STRING,
-            "GSSAPI",
-            ConfigDef.Importance.MEDIUM,
-            KAFKACLIENT_SASL_MECHANISM_DOC
-        )
-        .define(
-            KAFKACLIENT_SASL_KERBEROS_KINIT_CMD_CONFIG,
-            ConfigDef.Type.STRING,
-            "/usr/bin/kinit",
-            ConfigDef.Importance.LOW,
-            KAFKACLIENT_SASL_KERBEROS_KINIT_CMD_DOC
-        )
-        .define(
-            KAFKACLIENT_SASL_KERBEROS_MIN_TIME_BEFORE_RELOGIN_CONFIG,
-            ConfigDef.Type.LONG,
-            60000,
-            ConfigDef.Importance.LOW,
-            KAFKACLIENT_SASL_KERBEROS_MIN_TIME_BEFORE_RELOGIN_DOC
-        )
-        .define(
-            KAFKACLIENT_SASL_KERBEROS_TICKET_RENEW_JITTER_CONFIG,
-            ConfigDef.Type.DOUBLE,
-            0.05,
-            ConfigDef.Importance.LOW,
-            KAFKACLIENT_SASL_KERBEROS_TICKET_RENEW_JITTER_DOC
-        )
-        .define(
-            KAFKACLIENT_SASL_KERBEROS_TICKET_RENEW_WINDOW_FACTOR_CONFIG,
-            ConfigDef.Type.DOUBLE,
-            0.8,
-            ConfigDef.Importance.LOW,
-            KAFKACLIENT_SASL_KERBEROS_TICKET_RENEW_WINDOW_FACTOR_DOC
-        )
-        .define(
-            KAFKA_REST_RESOURCE_EXTENSION_CONFIG,
-            Type.LIST,
-            "",
-            Importance.LOW,
-            KAFKA_REST_RESOURCE_EXTENSION_DOC
-        );
+    return baseConfigDef(
+        KAFKAREST_PORT_DEFAULT,
```

Comment:
not directly related, but as we're working on these improvements -- thoughts on deprecating all the `PORT` stuff and committing to removing on next major version bump? we can mark the deprecations for removal in that upcoming major version bump.

Reply:
wilco. will add `@Deprecated` everywhere